server/index.js

```diff
@@ -63,6 +62,105 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart/', (req, res) => {
+  const sql = `
+    SELECT "c"."cartItemId",
+            "c"."price",
+            "p"."productId",
+            "p"."image",
+            "p"."name",
+            "p"."shortDescription"
+        FROM "cartItems" AS "c"
+        JOIN "products" AS "p" USING ("productId")
+      WHERE "c"."cartId" = $1
+  `;
+  const value = [req.session.cartId];
+  db.query(sql, value)
+    .then(result => {
+      res.json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/cart/', (req, res, next) => {
+  const { productId, price } = req.body;
+  if (!Number.isInteger(productId) || productId <= 0) {
+    next(new ClientError('"productId" must be a positive integer', 400));
+  }
+  const sqlInitialQuery = `
+    SELECT "productId"
+    FROM "products"
+    WHERE "productId" = $1
+  `;
+  const valueInitial = [productId];
+  db.query(sqlInitialQuery, valueInitial)
+    .then(result => {
+      if (!result.rows.length) {
+        res.status(400).json({
+          error: `Cannot find the productId:" ${productId}`
+        });
```

Comment:
You should `throw` a `ClientError` here so that the promise chain shortcuts to the `.catch()` callback.

Reply:
Fixed!